packages/rpc-methods/src/restricted/invokeSnap.ts

```diff
@@ -94,19 +131,23 @@ export function getInvokeSnapImplementation({
   return async function invokeSnap(
     options: RestrictedMethodOptions<Record<string, Json>>,
   ): Promise<Json> {
-    const { params = {}, method, context } = options;
+    const { params = {}, context } = options;
+
+    const { snapId, request } = params as InvokeSnapParams;
```

Comment:
Where do we enforce the permission? e.g. checking if the caveat exists for the specific snap we are trying to talk to?

Reply:
I think we use decorators for that in the entropy RPC requests

Reply:
I suppose currently, it's enforced implicitly through the `getSnap` check, so I choose to keep that as is. You are right that I do need to add a decorator property, it's required when restricted methods are called. Not sure what to have inside the decorator function though apart from just calling the method implementation...

Reply:
With the wildcard permission it is directly enforced by the permission system. Since you cannot call the restricted method without already having the permission. With the new permission a DApp could have the `wallet_snap` permission, but not for the specific snap trying to be invoked.

We should probably just mirror this implementation from `getBip44Entropy`: https://github.com/MetaMask/snaps-monorepo/blob/main/packages/rpc-methods/src/restricted/getBip44Entropy.ts#L182-L191
Meaning we should validate if the caveats on the permission are sufficient to communicate with the requested snap.